### PR TITLE
Prepare for Revised Implementation of WBPn

### PR DESCRIPTION
**Pull request by bska**
This initial commit changes the Summary class's API for consuming block-averaged well level pressure values (summary keywords WBPn). The former approach was intended to consume a collection of source values&mdash;pressures, densities, and pore-volumes&mdash;and then to defer calculation of the WBPn summary vectors to the Summary class.

This commit introduces a `WellBlockAvgPress` container class which holds precomputed WBPn results and the intention is to move the calculation to the simulator side for greater parallelism.

### CODE REVIEW

**Comment by bska** [comment]
jenkins build this opm-simulators=4691 please

File: src/opm/output/eclipse/Summary.cpp [hunk 1]
```diff
@@ -3502,22 +3503,29 @@ namespace Evaluator {
 
     std::string Factory::functionUnitString() const
     {
+        const auto unit_string_tracer = this->es_.tracer()
+            .get_unit_string(this->es_.getUnits(),
+                             this->node_->keyword);
+
+        if (unit_string_tracer != "") {
```
Comment (akva2) [inline src/opm/output/eclipse/Summary.cpp:3510]: .empty()

**Review by akva2 — APPROVED** [review]
Looks good to me.

File: src/opm/output/eclipse/Summary.cpp (hunk 1 above)
Comment (bska) [reply src/opm/output/eclipse/Summary.cpp]: > .empty()

Good point.  I've pushed an update to that effect.

**Comment by bska** [comment]
jenkins build this opm-simulators=4691 please

**Comment by bska** [comment]
jenkins build this opm-simulators=4691 please

**Comment by bska** [comment]
Merging this per prior approval.

**Pull request merged**